src/IonEventStream.ts

```diff
@@ -50,39 +50,43 @@ export class IonEventStream {
             writer.setAnnotations(tempEvent.annotations);
             switch (tempEvent.eventType) {
                 case IonEventType.SCALAR:
-                    switch (tempEvent.ionType) {
-                        case IonTypes.BOOL :
-                            writer.writeBoolean(tempEvent.ionValue);
-                            break;
-                        case IonTypes.STRING:
-                            writer.writeString(tempEvent.ionValue);
-                            break;
-                        case IonTypes.SYMBOL :
-                            writer.writeSymbol(tempEvent.ionValue);
-                            break;
-                        case IonTypes.INT :
-                            writer.writeInt(tempEvent.ionValue);
-                            break;
-                        case IonTypes.DECIMAL :
-                            writer.writeDecimal(tempEvent.ionValue);
-                            break;
-                        case IonTypes.FLOAT :
-                            writer.writeFloat64(tempEvent.ionValue);
-                            break;
-                        case IonTypes.NULL :
-                            writer.writeNull(tempEvent.ionType);
-                            break;
-                        case IonTypes.TIMESTAMP :
-                            writer.writeTimestamp(tempEvent.ionValue);
-                            break;
-                        case IonTypes.CLOB :
-                            writer.writeClob(tempEvent.ionValue);
-                            break;
-                        case IonTypes.BLOB :
-                            writer.writeBlob(tempEvent.ionValue);
-                            break;
-                        default :
-                            throw new Error("unexpected type: " + tempEvent.ionType!.name);
+                    if (tempEvent.ionValue == null) {
```

Comment:
* lifting the null check out becomes:
```js
if (tempEvent.eventType == IonEventType.SCALAR && tempEvent.ionValue == null) {
    writer.writeNull(tempEvent.ionType!);
} else {
    switch (tempEvent.eventType) {
        ...
    }
}
```
This doesn't end up reducing indentation, and I don't think it addresses the null container / scalar concern you mention.  I'm more inclined to leave as-is; let me know if I've misunderstood something.

* Interesting.  `tempEvent.ionValue` is of type `any`.  Not sure why the compiler allows this; perhaps because we haven't enabled strict checking?

Reply:
Ah, of course!  Thank you, opened https://github.com/amzn/ion-js/pull/578 to address this